### PR TITLE
framework/eventloop: Implement loop per task

### PR DESCRIPTION
**Pull request by jeongarmy**
- In default, a task has its own loop.
  User can add and handle the event in its own loop.

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [cbfbb5dc2934f1b143d1ea1ad3201f9952b4e305] - Code Rule Check Result:
framework/include/eventloop/eventloop.h:74: ERROR: [IDT_M_TAB] code indent should use tabs where possible
framework/include/eventloop/eventloop.h:75: ERROR: [IDT_M_TAB] code indent should use tabs where possible
framework/include/eventloop/eventloop.h:76: ERROR: [IDT_M_TAB] code indent should use tabs where possible
framework/include/eventloop/eventloop.h:103: ERROR: [IDT_M_TAB] code indent should use tabs where possible
framework/include/eventloop/eventloop.h:104: ERROR: [IDT_M_TAB] code indent should use tabs where possible
framework/include/eventloop/eventloop.h:105: ERROR: [IDT_M_TAB] code indent should use tabs where possible
framework/src/eventloop/eventloop_task.c:53: ERROR: [SPC_M_KWD] space required before the open parenthesis '('

**Comment by seinfra** [comment]
Target : [1dd89befa2c2c875dcf60f7382552640d227963b] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [1dd89befa2c2c875dcf60f7382552640d227963b] - Code Rule Check (C++) OK.

File: framework/src/eventloop/eventloop_loop.c [hunk 1]
```diff
@@ -18,21 +18,82 @@
...
 {
-	uv_run(uv_default_loop(), UV_RUN_DEFAULT);
+	el_loop_t *loop;
+
+	loop = get_app_loop();
+	if (loop == NULL) {
+		return;
+	}
+
+	uv_run(loop, UV_RUN_DEFAULT);
```
Comment (davidfather) [inline framework/src/eventloop/eventloop_loop.c]: Why don't you add a return value to check whether the loop is successfully started?

File: framework/src/eventloop/eventloop_timer.c [hunk 2]
```diff
@@ -59,10 +62,10 @@ static void timeout_callback_func(el_timer_t *timer)
...
 
-el_timer_t *eventloop_add_timer(unsigned int timeout, bool repeat, timeout_callback func, void *data)
+static el_timer_t *add_timer(el_loop_t *loop, unsigned int timeout, bool repeat, timeout_callback func, void *data)
 {
-	el_timer_t *timer = NULL;
-	timer_cb_t *callback = NULL;
+	el_timer_t *timer;
+	timer_cb_t *callback;
 
 	timer = (el_timer_t *)EL_ALLOC(sizeof(el_timer_t));
```
Comment (davidfather) [inline framework/src/eventloop/eventloop_timer.c]: Why don't you check whether the input parameters are valid before any memory allocation?

File: os/kernel/init/os_bringup.c [hunk 3]
```diff
@@ -263,17 +263,26 @@ static inline void os_do_appstart(void)
 	}
 #endif
 
+#ifdef CONFIG_EVENTLOOP
+#define EVENTLOOP_STACK_SIZE 4096
+#define EVENTLOOP_PRIORITY 201
+	svdbg("Starting eventloop default loop\n");
+	pid = task_create("eventloop", EVENTLOOP_PRIORITY, EVENTLOOP_STACK_SIZE, eventloop_task, (FAR char *const *)NULL);
```
Comment (sunghan-chang) [inline os/kernel/init/os_bringup.c]: Let's move this to init.c preapp_init()

**Comment by seinfra** [comment]
Target : [9c00e9635d2b32d288f82da912fb12af18ec3427] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [9c00e9635d2b32d288f82da912fb12af18ec3427] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [3a607a574cf4d3ae1de98089bf831a53de5f51fd] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [3a607a574cf4d3ae1de98089bf831a53de5f51fd] - Code Rule Check (C++) OK.

File: os/kernel/init/os_bringup.c [hunk 4]
```diff
@@ -266,14 +266,13 @@ static inline void os_do_appstart(void)
...
-	task_manager_drv_register();
-
-	pid = task_create("task_manager", TASKMGR_PRIORITY, TASKMGR_STACK_SIZE, task_manager, (FAR char *const *)NULL);
-	if (pid < 0) {
-		svdbg("Failed to create Task Manager\n");
-	}
+		pid = task_create("task_manager", TASKMGR_PRIORITY, TASKMGR_STACK_SIZE, task_manager, (FAR char *const *)NULL);
+		if (pid < 0) {
+			svdbg("Failed to create Task Manager\n");
+		}
```
Comment (lokeshbv) [inline os/kernel/init/os_bringup.c]: please check alignment

**Comment by seinfra** [comment]
Target : [13e96bb6526c98928a996ba9ec040a291970aee5] -

**Comment by seinfra** [comment]
Target : [2348e9a9b8bcafa02255e13c62087db9b8e514a2] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [2348e9a9b8bcafa02255e13c62087db9b8e514a2] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [3ff7e7c9c0d4d81af3e4249e4a1ccafcbf76bd8c] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [afc1860040eddc6088aa422b8adac0a37dc2c52d] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [afc1860040eddc6088aa422b8adac0a37dc2c52d] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [f4c7d984396c5b8d04dad6d0d8bd8aff25a9d61d] -

**Comment by seinfra** [comment]
Target : [f4c7d984396c5b8d04dad6d0d8bd8aff25a9d61d] -

**Pull request merged**